### PR TITLE
Added addon config to cluster config

### PR DESCRIPTION
**Pull request by thorfour**

### CODE REVIEW

File: modules/tectonic/resources/manifests/cluster-config.yaml [hunk 1]
```diff
@@ -29,3 +29,9 @@ data:
       installerPlatform: ${platform}
       kubeAPIserverURL: ${kube_apiserver_url}
       tectonicVersion: ${tectonic_version}
+  addon-config:
+      apiVersion: v1
+      kind: addonConfig
```
Comment (rithujohn191) [inline modules/tectonic/resources/manifests/cluster-config.yaml]: I think this needs to be `CamelCase`. I think this is a Kubernetes standard

File: modules/tectonic/resources/manifests/cluster-config.yaml [hunk 2]
```diff
@@ -29,3 +29,9 @@ data:
       installerPlatform: ${platform}
       kubeAPIserverURL: ${kube_apiserver_url}
       tectonicVersion: ${tectonic_version}
+  addon-config:
```
Comment (yifan-gu) [inline modules/tectonic/resources/manifests/cluster-config.yaml]: maybe adding `|` to keep consistency?

File: modules/tectonic/resources/manifests/cluster-config.yaml [hunk 3]
```diff
@@ -29,3 +29,9 @@ data:
...
+      apiVersion: v1
+      kind: AddonConfig
+      heapsterConfig:
+      dnsConfig:
```
Comment (yifan-gu) [inline modules/tectonic/resources/manifests/cluster-config.yaml]: ditto

File: modules/tectonic/resources/manifests/cluster-config.yaml (hunk 2 above)
Comment (thorfour) [reply modules/tectonic/resources/manifests/cluster-config.yaml]: Isn't the | unnecessary?

**Comment by sym3tri** [comment]
Hey @thorfour @yifan-gu, this PR broke master. Can you please run smoke tests before merging any changes to shared resources?

**Pull request merged**